token/services/db/sql/common/tokens.go

```diff
@@ -186,6 +186,37 @@ func (db *TokenDB) SpendableTokensIteratorBy(ctx context.Context, walletID strin
 	return &UnspentTokensInWalletIterator{txs: rows}, nil
 }
 
+// UnspendableTokensIteratorBy returns the minimum information for conversion about the tokens that cannot be spent
+func (db *TokenDB) UnspendableTokensIteratorBy(ctx context.Context, walletID string, tokenType token.Type) (tdriver.UnspendableTokensIterator, error) {
+	// first select all the distinct ledger types
+	includeFormats, err := db.unspendableTokenFormats(ctx, walletID, tokenType)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get unspendable token formats")
+	}
+	logger.Debugf("after filtering we have [%v]", includeFormats)
+
+	span := trace.SpanFromContext(ctx)
+	// now, select the tokens with the list of ledger tokens
+	where, args := common.Where(db.ci.HasTokenDetails(driver.QueryTokenDetailsParams{
+		WalletID:           walletID,
+		TokenType:          tokenType,
+		Spendable:          driver.Any,
```

Comment:
For the sake of completeness if we're querying all unspendable tokens with this walletID and tokenType the condition should be

```
WHERE wallet_id = ... AND token_type = ... AND (spendable == true OR format IN (...))
```
Otherwise maybe we could rename this method to `UnspendableFormatTokensIteratorBy`